integration_tests/suite/base/test_registrars.py

```diff
@@ -0,0 +1,460 @@
+    yield s.check_bogus_field_returns_error, url, 'proxy_backup_port', 'wrong'
+    yield s.check_bogus_field_returns_error, url, 'outbound_proxy_host', 123
+    yield s.check_bogus_field_returns_error, url, 'outbound_proxy_port', 'wrong'
+
+
+@fixtures.registrar(
+    id='VisibleId',
+    name='VisibleRegistrar',
+    proxy_main_host='1.2.3.4',
+    outbound_proxy_port=5060,
+
+)
+@fixtures.registrar(
+    id='HiddenId',
+    name='HiddenRegistrar',
+    proxy_main_host='1.2.2.4',
+    outbound_proxy_port=5061,
+)
+def test_search(registrar, hidden):
+    url = confd.registrars
+    searches = {
+        'proxy_main_host': '2.3',
+        'id': 'leid',
+        'name': 'ereg',
+        'outbound_proxy_port': 5060,
+    }
+
+    for field, term in searches.items():
+        yield check_search, url, registrar, hidden, field, term
+
+
+def check_search(url, registrar, hidden, field, term):
+    response = url.get(search=term)
+    assert_that(response.items, has_item(has_entry(field, registrar[field])))
+    assert_that(response.items, is_not(has_item(has_entry(field, hidden[field]))))
+
+    response = url.get(**{field: registrar[field]})
+    assert_that(response.items, has_item(has_entry('id', registrar['id'])))
+    assert_that(response.items, is_not(has_item(has_entry('id', hidden['id']))))
+
+
+@fixtures.registrar(
+    proxy_main_host="99.20.30.40",
+    name="SortRegistrar1",
+    main_port=5060,
+)
+@fixtures.registrar(
+    proxy_main_host="99.21.30.40",
+    name="SortRegistrar2",
+    main_port=5061,
+)
+def test_sorting_offset_limit(registrar1, registrar2):
+    url = confd.registrars.get
+    yield s.check_sorting, url, registrar1, registrar2, 'proxy_main_host', '99.'
+    yield s.check_sorting, url, registrar1, registrar2, 'name', 'SortRegistrar'
+    yield s.check_sorting, url, registrar1, registrar2, 'main_port', 'SortRegistrar'
+
+
+@fixtures.registrar()
+def test_get(registrar):
+    response = confd.registrars(registrar['id']).get()
+    assert_that(response.item, has_entries(**registrar))
+
+
+def test_create_registrar_minimal_parameters():
+    response = confd.registrars.post(main_host='1.2.3.4', proxy_main_host='1.2.3.4')
+    response.assert_created('registrars')
+
+    assert_that(
+        response.item,
+        has_entries(
+            name=none(),
+            main_host='1.2.3.4',
+            main_port=none(),
+            backup_host=none(),
+            backup_port=none(),
+            proxy_main_host='1.2.3.4',
+            proxy_main_port=none(),
+            proxy_backup_host=none(),
+            proxy_backup_port=none(),
+            outbound_proxy_host=none(),
+            outbound_proxy_port=none(),
+        )
+    )
+
+
+def test_create_registrar_null_parameters():
+    response = confd.registrars.post(
+        main_host='1.2.3.4',
+        proxy_main_host='1.2.3.4',
+        main_port=None,
+        name=None,
+        outbound_proxy_host=None,
+        outbound_proxy_port=None,
```

Comment:
Why not setting all possible parameters to none? But I must admit that test is a little bit useless